app/components/primer/base_component.rb

```diff
@@ -152,6 +150,35 @@ def initialize(tag:, classes: nil, **system_arguments)
 
       raise ArgumentError, "`class` is an invalid argument. Use `classes` instead." if system_arguments.key?(:class) && !Rails.env.production?
 
+      if (denylist = system_arguments[:system_arguments_denylist])
+        if Rails.application.config.primer_view_components.force_system_arguments && !ENV["PRIMER_WARNINGS_DISABLED"]
+          # Convert denylist from:
+          # { [:p, :pt] => "message" } to:
+          # { p: "message", pt: "message" }
+          unpacked_denylist =
+            denylist.each_with_object({}) do |(keys, value), memo|
+              keys.each { |key| memo[key] = value }
+
+              memo
```

Comment:
```suggestion
              keys.each { |key| memo[key] = value }
```

I'm pretty sure that you don't need to return `memo` in `each_with_object`